sdk/main/src/AccountBalanceQuery.cc

```diff
@@ -21,113 +21,52 @@
 
 #include "AccountBalance.h"
 
-#include <proto/basic_types.pb.h>
 #include <proto/crypto_get_account_balance.pb.h>
 #include <proto/query.pb.h>
-#include <proto/query_header.pb.h>
 #include <proto/response.pb.h>
 
 namespace Hedera
 {
 //-----
-AccountBalanceQuery::AccountBalanceQuery()
-  : Query()
-  , mAccountId()
-  , mContractId()
+AccountBalanceQuery& AccountBalanceQuery::setAccountId(const AccountId& accountId)
 {
-}
-
-//-----
-AccountBalanceQuery::AccountBalanceQuery(const AccountId& accountId)
-  : Query()
-  , mContractId()
-{
-  mAccountId.setValue(accountId);
-}
+  mAccountId = accountId;
+  mContractId.reset();
 
-//-----
-AccountBalanceQuery::AccountBalanceQuery(const ContractId& contractId)
-  : Query()
-  , mAccountId()
-{
-  mContractId.setValue(contractId);
+  return *this;
 }
 
 //-----
-void
-AccountBalanceQuery::onMakeRequest(proto::Query* query,
-                                   proto::QueryHeader* header) const
+AccountBalanceQuery& AccountBalanceQuery::setContractId(const ContractId& contractId)
 {
-  proto::CryptoGetAccountBalanceQuery* getAccountBalanceQuery =
-    query->mutable_cryptogetaccountbalance();
+  mContractId = contractId;
+  mAccountId.reset();
 
-  if (mAccountId.isValid())
-  {
-    getAccountBalanceQuery->set_allocated_accountid(
-      mAccountId.getValue().toProtobuf());
-  }
-
-  if (mContractId.isValid())
-  {
-    getAccountBalanceQuery->set_allocated_contractid(
-      mContractId.getValue().toProtobuf());
-  }
-
-  getAccountBalanceQuery->set_allocated_header(header);
+  return *this;
 }
-
 //-----
-proto::ResponseHeader
-AccountBalanceQuery::mapResponseHeader(proto::Response* response) const
+proto::Query AccountBalanceQuery::makeRequest() const
 {
-  return response->cryptogetaccountbalance().header();
-}
+  proto::Query query;
+  proto::CryptoGetAccountBalanceQuery* getAccountBalanceQuery = query.mutable_cryptogetaccountbalance();
 
-//-----
-void
-AccountBalanceQuery::validateChecksums(const Client& client) const
-{
-  if (mAccountId.isValid())
+  if (mAccountId)
```

Comment:
Should the mutual exclusivity of `mAccountId` and `mContractId` be reflected here somehow?

Reply:
I don't think it needs to be, since the set functions prevent both variables being set, unless there's a corner case I'm missing.